ReactAndroid/src/main/java/com/facebook/react/bridge/BaseJavaModule.java

```diff
@@ -49,17 +54,35 @@
  */
 public abstract class BaseJavaModule implements NativeModule {
   // taken from Libraries/Utilities/MessageQueue.js
-  static final public String METHOD_TYPE_ASYNC = "async";
-  static final public String METHOD_TYPE_PROMISE= "promise";
+  private static final String METHOD_TYPE_ASYNC = "async";
+  private static final String METHOD_TYPE_PROMISE= "promise";
   static final public String METHOD_TYPE_SYNC = "sync";
+  private final List<? extends ArgumentExtractor.Factory> mArgumentExtractorFactories;
+
+  public BaseJavaModule() {
+    this(Collections.<ArgumentExtractor.Factory>emptyList());
+  }
 
-  private static abstract class ArgumentExtractor<T> {
+  public BaseJavaModule(List<? extends ArgumentExtractor.Factory> argumentExtractorFactories) {
```

Comment:
I though about doing this way but I'm really not a fan of having this "inheritance party" behavior. Using constructor arguments is a much more explicit way of letting subclasses know about that functionality and having a hook point for defining that.
I understand if you're not a fan of that style though and I'd be happy to change it